src/lib/source-handlers/azure/list-projects.ts

```diff
@@ -0,0 +1,95 @@
+import * as qs from 'querystring';
+import base64 = require('base-64');
+import Bottleneck from 'bottleneck';
+import * as needle from 'needle';
+import * as debugLib from 'debug';
+import { AzureProjectData } from './types';
+import { getAzureToken } from './get-azure-token';
+import { getBaseUrl } from './get-base-url';
+import { AzureProjectData as Project } from './types';
+
+const debug = debugLib('snyk:azure');
+
+const limiter = new Bottleneck({
+  maxConcurrent: 1,
+  minTime: 500,
+});
+
+limiter.on('failed', async (error, jobInfo) => {
+  const id = jobInfo.options.id;
+  debug(`Job ${id} failed: ${error}`);
+  if (jobInfo.retryCount === 0) {
+    // Here we only retry once
+    debug(`Retrying job ${id} in 25ms!`);
+    return 25;
+  }
+});
+
+async function fetchAllProjects(orgName: string, host?: string): Promise<Project[]> {
+  const projectList: Project[] = [];
+  let hasMorePages = true;
+  let displayPageNumber = 1;
+  let nextPage: string | undefined;
+  while (hasMorePages) {
+    debug(`Fetching page ${displayPageNumber} for Azure projects`);
+    const { projects, continueFrom } = await getProjects(orgName, host, nextPage);
+    if (continueFrom && projects.length > 0) {
+      nextPage = continueFrom;
+      displayPageNumber++;
+    } else {
+      hasMorePages = false;
+    }
+    projects.map((project: { name: string; id: string }) => {
```

Comment:
There is no need to type project again, it is already typed from the function getProjects.

Reply:
Done